internal/provider/data_source_pullzone.go

```diff
@@ -0,0 +1,86 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourcePullZone() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePullZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"pull_zone_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			keyAWSSigningKey: {
+				Type:        schema.TypeString,
+				Description: "AWS Signing Key",
+				Computed:    true,
+			},
+			keyAWSSigningRegionName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The AWS Signing region name.",
+			},
+			keyAWSSigningSecret: {
+				Type:        schema.TypeString,
+				Sensitive:   true,
+				Computed:    true,
+				Description: "The AWS Signing region secret.",
```

Comment:
What is the usecase for having the AWS* fields in the datasource?

Reply:
I personally don't have any, but I assumed that a common use case for a datasource would be to read secret keys and pass them around. I  can remove if you don't think that's likely.